composer/datasets/in_context_learning_evaluation.py

```diff
@@ -85,582 +109,764 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
 
-        Randomly select `num_fewshot` samples from the dataset (not including the sample at `sample_idx`) and format
-        them each as follows `{example_delimiter}{question_prelimiter}{context}{continuation_delimiter}{chain_of_thought}{cot_delimiter}{answer}`.
+    def check_defaults_are_set(self, dict_of_defaults: dict) -> None:
+        if all(v for v in dict_of_defaults.values()):
+            return
+        raise ValueError(
+            f"{type(self).__name__} missing required variable(s): {', '.join([k for k, v in dict_of_defaults.items() if not v])}"
+        )
 
-        `chain_of_thought` will default to empty if not present in the dataset but `context` and `answer` must be present.
+    def _read_dataset(self,
+                      dataset_uri: str,
+                      destination_path: str,
+                      hf_loading_vars: Dict = None,
+                      hf_parsing_map: Dict = None) -> transformers.Dataset:
+        """
+        Reads a dataset and handles parsing it from HuggingFace.
+        Args:
+            dataset_uri (str): A local path, a remote path beginning with ``s3://`` or another backend, or a HuggingFace dataset uri.
+                Alternate backends must be supported by :meth:`composer.utils.maybe_create_object_store_from_uri`.
+            destination_path (str): A local path where the data will be stored
+            hf_loading_vars (Dict): If parsing from HuggingFace, keyword args that will be passed into load_dataset
+            hf_parsing_map (Dict): Dictionary in the form of {icl_key: [hf_col1, hf_col2]} that will map one or more hf columns, in order, to ICL dataset columns
 
-        Returns the formatted prompt_string + concatenated list of formatted few shot examples.
+        Returns:
+            dataset: a loaded HF dataset
         """
-        prompt_and_fewshot = prompt_string
+        try:
+            from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
+        except ImportError as e:
+            raise MissingConditionalImportError(
+                extra_deps_group='nlp',
+                conda_package='datasets',
+                conda_channel='conda-forge',
+            ) from e
+        if _check_if_huggingface_uri(dataset_uri):
+            dataset = load_dataset(dataset_uri, **hf_loading_vars)
+            if hf_parsing_map:
+                dataset_parsing_func = lambda example: {
+                    k: ' '.join([str(example[col]) for col in v]) for k, v in hf_parsing_map.items()
+                }
+                dataset = dataset.map(dataset_parsing_func, remove_columns=dataset.column_names)
+        else:
+            with dist.local_rank_zero_download_and_wait(destination_path):
+                if dist.get_local_rank() == 0:
+                    get_file(dataset_uri, destination_path, overwrite=True)
+            dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
```

Comment:
We should have a way to handle arbitrary splits as well as datasets which include subsets, e.g. `dataset = load_dataset("tasksource/bigbench",'movie_recommendation')`

Reply:
lets defer this to a later feature request so we can get this PR in and build on it

Reply:
If you load from huggingface directly you can already do this with `name: movie_recommendation` in `hf_loading_vars` in your yaml

Reply:
Can you change that to "subset" since I believe that's what it is called on the HF hub?

Reply:
`dataset = load_dataset("tasksource/bigbench", name='movie_recommendation')` and `dataset = load_dataset("tasksource/bigbench", 'movie_recommendation')` both worked for me but `dataset = load_dataset("tasksource/bigbench", subset='movie_recommendation')` gave me `TypeError: bigbench_Config.__init__() missing 1 required positional argument: 'text_features'`